code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,276 @@
+
+/mob/living/simple_animal/hostile/eldritch/armsy/New(spawn_more = TRUE,len = 6)
+	. = ..()
+	oldloc = loc
+	if(!spawn_more)
+		return
+	allow_pulling = TRUE
+	var/mob/living/simple_animal/hostile/eldritch/armsy/next
+	var/mob/living/simple_animal/hostile/eldritch/armsy/prev
+	var/mob/living/simple_animal/hostile/eldritch/armsy/current
+	for(var/i = 0, i <= len,i++)
+		prev = current
+		//i tried using switch, but byond is really fucky and it didnt work as intended. Im sorry
+		if(i == 0)
+			current = new type(drop_location(),spawn_more = FALSE)
+			current.icon_state = "armsy_mid"
+			current.icon_living = "armsy_mid"
+			current.front = src
+			current.AIStatus = AI_OFF
+			back = current
+		if(i > 0 && i < len)
+			current = new type(drop_location(),spawn_more = FALSE)
+			prev.back = current
+			prev.icon_state = "armsy_mid"
+			prev.icon_living = "armsy_mid"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		if(i == len)
+			prev.icon_state = "armsy_end"
+			prev.icon_living = "armsy_end"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		next = prev
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Moved()
+
+	if(pulledby && !allow_pulling)
+		pulledby.stop_pulling()
+		loc = oldloc
+
+	if(back && back.loc != oldloc)
+		for(var/i = 0, i < max(get_dist(loc,back.loc),10),i++)
+			step_towards(back,oldloc)
+			if(loc == back.loc)
+				break
+		back.loc = oldloc//just in case
+
+	if(front && loc != front.oldloc)
+		for(var/i = 0, i < max(get_dist(loc,front.loc),10),i++)
+			step_towards(src,front.oldloc)
+			if(loc == front.loc)
+				break
+		loc = front.oldloc//just in case
+
+	oldloc = loc
+	gib_trail()
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/gib_trail()
+	if(back)
+		return
+	var/chosen_decal = pick(typesof(/obj/effect/decal/cleanable/blood/gibs))
+	var/obj/effect/decal/cleanable/blood/gibs/decal = new chosen_decal(drop_location())
+	decal.setDir(dir)
+	return
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Destroy()
+	if(front)
+		front.icon_state = "armsy_end"
+		front.icon_living = "armsy_end"
+	if(back)
+		back.Destroy() // chain destruction baby
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/heal()
+	if(health == maxHealth)
+		if(back)
+			back.heal()
+			return
+		else
+			current_stacks++
+			if(current_stacks >= stacks_to_grow)
+				var/mob/living/simple_animal/hostile/eldritch/armsy/prev = new type(drop_location(),spawn_more = FALSE)
+				icon_state = "armsy_mid"
+				icon_living =  "armsy_mid"
+				back = prev
+				prev.icon_state = "armsy_end"
+				prev.icon_living = "armsy_end"
+				prev.front = src
+				prev.AIStatus = AI_OFF
+
+				current_stacks = 0
+
+	adjustBruteLoss(-maxHealth/2)
+	adjustFireLoss(-maxHealth/2)
+	adjustToxLoss(-maxHealth/2)
+	adjustOxyLoss(-maxHealth/2)
+
+/mob/living/simple_animal/hostile/eldritch/armsy/AttackingTarget()
+	if(istype(target,/obj/item/bodypart/r_arm) || istype(target,/obj/item/bodypart/l_arm))
+		target.Destroy()
```

Comment:
Use `qdel`, not `Destroy()`
Remove every custom call of `Destroy()` in your PR.

Reply:
This is unresolved.

Reply:
god i missed one? fuck

Reply:
Why did you resolve this again without fixing it?

Reply:
aaaaaaaaaaaa